File: src/main/java/com/example/demo/mybatis/EncryptedStringTypeHandler.java
package com.example.demo.mybatis;

import com.example.demo.util.EncryptionUtils;
import org.apache.ibatis.type.BaseTypeHandler;
import org.apache.ibatis.type.JdbcType;

import java.sql.CallableStatement;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;

public class EncryptedStringTypeHandler extends BaseTypeHandler<String> {

    @Override
    public void setNonNullParameter(PreparedStatement ps, int i, String parameter, JdbcType jdbcType) throws SQLException {
        ps.setString(i, EncryptionUtils.encrypt(parameter));
    }

    @Override
    public String getNullableResult(ResultSet rs, String columnName) throws SQLException {
        return EncryptionUtils.decrypt(rs.getString(columnName));
    }

    @Override
    public String getNullableResult(ResultSet rs, int columnIndex) throws SQLException {
        return EncryptionUtils.decrypt(rs.getString(columnIndex));
    }

    @Override
    public String getNullableResult(CallableStatement cs, int columnIndex) throws SQLException {
        return EncryptionUtils.decrypt(cs.getString(columnIndex));
    }
}

File: src/main/kotlin/com/example/demo/mybatis/EncryptedStringTypeHandler.kt
package com.example.demo.mybatis

import com.example.demo.util.EncryptionUtils
import org.apache.ibatis.type.BaseTypeHandler
import org.apache.ibatis.type.JdbcType
import java.sql.CallableStatement
import java.sql.PreparedStatement
import java.sql.ResultSet

class EncryptedStringTypeHandler : BaseTypeHandler<String>() {
    override fun setNonNullParameter(
        ps: PreparedStatement,
        i: Int,
        parameter: String,
        jdbcType: JdbcType?
    ) {
        ps.setString(i, EncryptionUtils.encrypt(parameter))
    }

    override fun getNullableResult(rs: ResultSet, columnName: String): String? {
        return EncryptionUtils.decrypt(rs.getString(columnName))
    }

    override fun getNullableResult(rs: ResultSet, columnIndex: Int): String? {
        return EncryptionUtils.decrypt(rs.getString(columnIndex))
    }

    override fun getNullableResult(cs: CallableStatement, columnIndex: Int): String? {
        return EncryptionUtils.decrypt(cs.getString(columnIndex))
    }
}
